components/stacks-network/src/chains_coordinator.rs

```diff
@@ -810,6 +791,124 @@ pub async fn publish_stacking_orders(
     }
 }
 
+fn fund_genesis_account(
+    devnet_event_tx: &Sender<DevnetEvent>,
+    services_map_hosts: &ServicesMapHosts,
+    accounts: &[AccountConfig],
+    fee_rate: u64,
+    bitcoin_block_height: u32,
+    boot_completed: &Arc<AtomicBool>,
+) {
+    let deployer = accounts
+        .iter()
+        .find(|account| account.label == "deployer")
+        .unwrap()
+        .clone();
+    let (_, _, deployer_secret_key) = clarinet_files::compute_addresses(
+        &deployer.mnemonic,
+        &deployer.derivation,
+        &StacksNetwork::Devnet.get_networks(),
+    );
+
+    let accounts_moved = accounts.to_vec();
+    let devnet_event_tx_moved = devnet_event_tx.clone();
+    let stacks_api_host_moved = services_map_hosts.stacks_api_host.clone();
+    let boot_completed_moved = Arc::clone(boot_completed);
+
+    let _ = hiro_system_kit::thread_named("sBTC funding handler").spawn(move || {
+        while !boot_completed_moved.load(Ordering::SeqCst) {
+            std::thread::sleep(std::time::Duration::from_secs(3));
```

Comment:
Is this generally how long it takes? Seems like it wouldn't be an issue to pick a smaller sleep

Reply:
The boot itself takes between 1 and 2 minutes.

But this function is triggered after the sbtc-deposit contract is called (so at this point we are already in epoch 3.0).
At this point, `boot_completed` is likely false since there can be other contracts to be deployed (in 3.0, and 3.1).
It will takes a few blocks, so a few seconds, we check if it's done every 3 seconds.
But I can make the check every 1sec for a bit more reactivity 👍